### PR TITLE
[sidelining] add commented out harmlessness check

### PR DESCRIPTION
**Pull request by tomsilver**

### CODE REVIEW

File: src/nsrt_learning/strips_learning/gen_to_spec_learner.py [hunk 1]
```diff
@@ -86,6 +86,10 @@ def _learn(self) -> List[PartialNSRTAndDatastore]:
             # keep effects.
             self._induce_delete_side_keep(param_opt_to_nec_pnads)
 
+            # Harmlessness should now hold, but it's slow to check.
+            # assert self._check_harmlessness(
+            #     self._get_uniquely_named_nec_pnads(param_opt_to_nec_pnads))
```
Comment (tomsilver) [inline src/nsrt_learning/strips_learning/gen_to_spec_learner.py:91]: using `cur_itr_pnads_unfiltered` instead leads to a pyperplan error

File: src/nsrt_learning/strips_learning/gen_to_spec_learner.py [hunk 2]
```diff
@@ -345,8 +349,11 @@ def _get_uniquely_named_nec_pnads(
         uniquely_named_nec_pnads: List[PartialNSRTAndDatastore] = []
         for pnad_list in sorted(param_opt_to_nec_pnads.values(), key=str):
             for i, pnad in enumerate(pnad_list):
-                pnad.op = pnad.op.copy_with(name=(pnad.op.name + str(i)))
-                uniquely_named_nec_pnads.append(pnad)
+                new_op = pnad.op.copy_with(name=(pnad.op.name + str(i)))
```
Comment (tomsilver) [inline src/nsrt_learning/strips_learning/gen_to_spec_learner.py:352]: without this change, checking harmlessness modifies the PNAD name in place, leading to the name getting extended after each iteration

**Review by ronuchit — COMMENTED** [review]
lgtm, let @NishanthJKumar review

**Review by NishanthJKumar — APPROVED** [review]
LGTM! Thanks!!!

**Pull request merged**